GG-43272 Add vector similarity function

manini-zs opened the pull request.

manini-zs committed:
Added the complete similarity function implementation without score

manini-zs committed:
Handled the similarity function to default to cosine if an invalid function id is provided

manini-zs committed:
Fixed: A check to see if index type is vector and only then look for similarityFunction

manini-zs committed:
Fixed the Javadoc & Style issues

manini-zs committed:
Changed the order of the field to the end

manini-zs committed:
Merge branch 'master' into gg-43272-vector-similarity-function

manini-zs committed:
Removed the else, no similarity function will be set in else

manini-zs committed:
Removed the else, no similarity function will be set in else

manini-zs committed:
Fixed binary type ID inconsistency for QueryIndex with vector similarity function

Fixes: TestRemoteResultErrorNotMarshalable test failure

manini-zs committed:
Merge remote-tracking branch 'origin/master' into gg-43272-vector-similarity-function

ptupitsyn reviewed (changes requested):
Protocol changes look good to me.

manini-zs committed:
Moved the SimilarityFunction enum and updated java docs

Moved the SimilarityFunction enum to the cache package.
Updated java docs to reflect that the default value is COSINE
Also removed a redundant else

manini-zs committed:
Moved the boolean field out of the loop - required only once

manini-zs committed:
Made the SimilarityFunction enum consistent with QueryIndexType and other enums.

manini-zs committed:
Post using ordinal as the number for SimilarityFunction enum, the ordinal value that corresponds to COSINE becomes 0 and not 1

manini-zs committed:
Merge branch 'master' into gg-43272-vector-similarity-function
Merging the master

manini-zs committed:
Added the License / Copyright notice on top
